components/validation/form-element-mixin.js

```diff
@@ -0,0 +1,35 @@
+export const FormElementMixin = superclass => class extends superclass {
+
+	constructor() {
+		super();
+		this.__hiddenInput = document.createElement('input');
+		this.__hiddenInput.type = 'hidden';
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		requestAnimationFrame(() => {
+			const parentNode = this.parentNode;
+			const parent = parentNode.nodeType === Node.DOCUMENT_FRAGMENT_NODE ? this.getRootNode().host : parentNode;
```

Comment:
I think this would need to be `getRootNode({composed: true})`.  Without the `composed: true` (which defaults to `false`), `getRootNode` will return the shadow root, but we'd want to append these hidden form elements to the document root.